docs/enhancements/boot-status-reporter.md

```diff
@@ -0,0 +1,129 @@
+
+### Information Needed And Changes To The Discovery Agent
+
+In order to communicate with `assisted-service`, The assisted boot reported service needs to know:
+1. The `assisted-service` IP address and port.
+2. The pull-secret to use for agent authentication.
+3. The InfraEnv ID and host ID
+
+It will need this information starting from the first boot, which is a pointer ignition-based boot.
+
+For that, we need to modify the discovery agent to store this information on disk in a pre-defined location for the assisted boot reporter service to use.
+
+The assisted boot reporter service should start via `systemd` and determine its operating stage. As mentioned above:
+
+### First Boot (Pointer Ignition)
+Expected stage to be read from assisted-service backend: `Configuring`.
+Update the host stage to: `Booted with local ignition`.
+Start a timer to compress and send logs every 10 minutes to `v2UpdateHostInstallProgress` API endpoint.
+
+
+### Second Boot (Control Plane Ignition)
+Expected stage to be read from assisted-service backend: `Waiting for control plane`.
+Update the host stage to: `Booted with control plane ignition`.
+Start a timer to compress send logs every 10 minutes to `v2UpdateHostInstallProgress` API endpoint.
+Start another timer to halt operation and self-cleanup as follows:
+
+1. If 1 hour has passed.
+2. If cluster install progress reached a point where it is no longer needed (now counting in `assisted-controller` pod). An indication could be that `kubelet` is up and running.
+   Cleanup should include:
+
+   * The stored files for pull-secret, host id, and infraEnv id.
+   * Service itself - should not run on additional host boot.
+
+### Collected Information
+1. journalctl
+2. A list of mounted file systems
+3. crio logs
+4. network settings
+
+
+### Open Questions
+
+1. What should we use for implementation? Our options are:
+   - Bash:
+      - Advantages:
+          1. Has very little to no dependencies.
+          2. Expected to be present in future versions of RHCOS.
+      - Disadvantages:
+         1. Harder to maintain and test.
+         2. No code discovery agent code reuse (written in Go).
+         3. Client work will involve a lot of parsing.
+   - Go:
+      - Advantages:
+         1. Discovery agent code reuse (also written in Go).
+         2. Simpler to test and maintain, as we do for other assisted code bases. 
+         3. Can be shipped as a binary, which helps with dependencies.
+      - Disadvantages:
+         1. Not clear if we can include a Go binary in the ignition.
+
+   - Python:
+      - Advantages:
+         1. Code reuse to some extent. Swagger is able to generate a Python-based client. 
+         2. Simpler to test and maintain, comparing to bash.
+      - Disadvantages:
+         1. Python is not included with RHCOS. We will need to install it and some third-party python libs it will probably need (e.g., python-requests).
+
+
+2. How to handle pending user action state?
```

Comment:
`pending user action` might be the node current stage instead of `Rebooting` (the node moves to `pending user action` in case the node failed to boot from the installation disk.
So I think we should handle it the same way as we handle `Rebooting` if the `boot-reporter` service started it means that the node booted with the right disk and we can move to `Booted with control plane ignition`

Reply:
Is there a need to write anything to disk? we aim to upload logs that can be pulled in real-time (journal etc)